pkg/util/configspec.go

```diff
@@ -190,3 +192,35 @@ func EnsureMinHardwareVersionInConfigSpec(
 	}
 	configSpec.Version = configSpecHwVersion.String()
 }
+
+// SafeConfigSpecToString returns the string-ified version of the provided
+// ConfigSpec, first trying to use the special JSON encoder, then defaulting to
+// the normal JSON encoder.
+func SafeConfigSpecToString(
+	configSpec *vimtypes.VirtualMachineConfigSpec) (s string) {
+
+	if configSpec == nil {
+		return "null"
+	}
+
+	marshalWithNormalJSONEncoder := func() {
+		if data, err := json.Marshal(*configSpec); err != nil {
+			s = fmt.Sprintf("%v", *configSpec)
```

Comment:
@dougm Since you already reviewed this, I just wanted to highlight that I also made this change post-your review. This way, no matter what, there's *something* returned.